### DISCUSSION TITLE
Missing spectra while importing library(.msp)

### DISCUSSION THREAD

**Discussion by Jihyunchun71** [mzmine/mzmine#2038] [Q&A]
Hi, 
I'm using mzmine 4.2, and having problem with importing in-house library.
Our library is .msp format, and contains **567 spectra** (567 compounds with RT, precursor m/z, MS2 spectra, etc..).

However, whenever I import the library inside mzmine, **only 474 spectra appears.**
Is there any criteria of suitable spectra that can be imported as library spectra?

You can see the example of failed-to-be-imported spectra below. 

NAME: Compound A
RT: 12.30699
FORMULA: 
INCHIKEY:
SMILES:
PRECURSORMZ: 300.28958
Collision_energy: 40.0
Precursor_type: [M+H]+
Ion_mode: P
Num Peaks: 17
50.6662 0.00
55.0546 0.00
57.0703 36972292.00
62.0603 234070560.00
67.0543 0.00
69.0701 0.00
71.0856 26415518.00
81.0699 7684796.50
83.0856 0.00
85.1011 17483080.00
95.0855 12994498.00
109.1012 0.00
165.3823 0.00
207.4620 0.00
282.2800 10098412.00
283.2639 29053876.00
300.2902 0.00

I'd be grateful with any possible reasons or solutions.

Thanks,
Jihyun

**Comment by ansgarkorf** [comment]
Can you share a second spectrum that works?
How was the library generated?

**Comment by Jihyunchun71** [comment]
Hi, thank you for your kind reply.
This is one of the spectra that are successfully imported in mzmine.


NAME: Compound B
RETENTIONTIME: 5.00007
FORMULA: 
INCHIKEY:
SMILES:
PRECURSORMZ: 153.05475
COLLISIONENERGY: 30.00,40.00,50.00
PRECURSORTYPE: [M+H]+
IONMODE: Positive
Num Peaks: 13
65.0395 7299401.50
66.0428 226859.88
93.0341 21470474.00
94.0374 793861.81
97.0654 297975.19
110.0367 1934409.25
111.0445 17124680.00
112.0478 743306.50
125.0600 17900812.00
126.0633 740958.25
134.0603 942148.75
153.0547 7442925.00
154.0579 417669.88

My assumption is that if there are too many fragment spectra with zero intensity, the compound fails to be imported in library.
Is my assumption possible?

Thanks, 
Jihyun

**Comment by ansgarkorf** [comment]
Yes, I share your assumption. How was the library generated?

**Reply by Jihyunchun71** [reply]
The library is generated via Thermo mzvault, which is a library generating program provided by ThermoFisher.
We import MS2 spectra and other information of each compound to the program.
After all procedures including importing and curation are all done, we export the library as a .msp file.

**Reply by ansgarkorf** [reply]
thanks for explaining! To ensure best compatibility with mzmine you may want to checkout our library generation workflow for upcoming projects:
https://mzmine.github.io/mzmine_documentation/workflows/librarygeneration/library_generation.html

Please let us know if removing the 0 intensity values does solves your problem.

**Reply by Jihyunchun71** [reply]
Thank you for your advice.
After removing fragment spectra with 0 intensity, those spectra were successfully imported.

Plus, I noticed that compounds with several (not many) spectra with 0 intensity are well imported. 
I still don't know the threshold number or ratio of fragments with zero intensity that can be imported,
so it would be helpful if the threshold number were suggested in mzmine.

Also, by removing the 0 intensity values, the real number of fragment peaks doesn't match the original 'Num peaks = *' value. (Example below)

NAME: Compound C
RT: 4.76492
FORMULA: 
INCHIKEY:
SMILES:
PRECURSORMZ: 160.13330
Collision_energy: 40.0
Precursor_type: [M+H]+
Ion_mode: P
**Num Peaks: 18**
55.0547 46347.35
97.1012 30770.83
114.1279 586747.63


Since it is laborious to modify the 'Num peak' values one by one, can I simply remove the 'Num peaks =' entry in the library? Will this cause any issues, such as affecting the identification scoring process or something else?

Again, thank you for your kind support.
